design/helm.md

```diff
@@ -0,0 +1,145 @@
+# Helm Addons
+
+To be an addon in a Kurl cluster, they must be contributed into the repo and maintained across new breaking changes and new versions.
+This proposal will give users the flexibility to specify their own addons through a helm chart support in the Kurl installer spec.
+
+
+## Goals
+- Allow any Helm chart to be specified as part of a Kurl spec.
+
+## Non Goals
+
+- Verify authenticity, trustworthyness and "goodness" of helm addons.
+- Validate Helm values.
+
+## Background
+
+Users of Kurl have a limited set of available cluster addons and often want to include ones that are unsupported.
+There is friction in the [kurl development process](https://github.com/replicatedhq/kURL/blob/master/test/dev.md), [contributing new addons](https://kurl.sh/docs/add-on-author/) and addon maintanence. 
+
+Helm is a popular tool for managing deployment configuration that allows application/addon developers or the community to maintain a consistent interface for deployment through a chart.
+By allowing Kurl users to specifify a chart and its values, we can allow customization of kutl with new addons while delegating addons support and maintance to its community.
+
+## High-Level Design
+
+Kurl will bundle the Helm binary into the common tools bundle already delivering krew and kubectl plugins. The install.sh script will perform the following in the online case:
+1. Unarchive Helm and move into the `$PATH` and `/usr/bin`
+1. For each release, check if the release exists. 
+1. If the release exists, perform a `helm upgrade`.
+1. If the release doesn't exist, perform a `helm install`
+
+For airgapped the chart and artifacts need to be packaged into a bundle. The builder should perform the following:
+1. `helm pull` the repo
+1. `helm template` to render the chart with the provided values
```

Comment:
could i include an `installer-spec-file` as the end customer to override values?

Reply:
Per our conversation, I'm not sure if we do (not) need this ability with helm and if it's desirable to have users change this from the install command vs. in the installer. I think your point, which I agree with, is that users were doing this to get around inflexible addons, and now they can modify values directly in the installer to make similar changes.

Reply:
My preference is to eventually let end-users provide their own kurl installer spec, which would then let them specify any Helm chart modifications too. I'm not sure we need to go half-way here and let end-users augment vendor-provided specs.

Reply:
@DanStough i want to clarify what "users" in your message means. i am assume you are referring to the vendor